### ISSUE TITLE
VS Code just deleted all my files

### ISSUE THREAD

**Issue by dbrae** [microsoft/vscode#196223]
Type: <b>Performance Issue</b>

This is an emergency. I clicked source control and initialize repository then it connected to my pc. I later hit reverse saved changes and it deleted all my files permanently. I need them recovered please I can't find them anywhere... obviously i never commited anything because i didnt want to commit my personal files to github and now I've lost it all

VS Code version: Code 1.83.1 (f1b07bd25dfad64b0167beb15359ae573aecd2cc, 2023-10-10T23:48:05.904Z)
OS version: Windows_NT x64 10.0.19045
Modes: Unsupported

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|AMD Ryzen 5 5600 6-Core Processor               (12 x 3500)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|undefined|
|Memory (System)|31.93GB (18.16GB free)|
|Process Argv|--crash-reporter-id f61b7fa5-ddc2-4c34-9f2b-5614a991840c|
|Screen Reader|no|
|VM|40%|
</details><details>
<summary>Process Info</summary>

```
CPU %	Mem MB	   PID	Process
    1	    99	 16836	code main
    0	   105	  1564	shared-process
    1	   215	  5672	extensionHost [1]
    0	    39	  8816	   utility-network-service
    0	    28	  9060	   crashpad-handler
    0	    98	 12320	   gpu-process
    0	    83	 19172	fileWatcher [1]
    0	   508	 20396	window [1] (version_2.py - recovery - Visual Studio Code)
    0	    89	 20528	window [2] (Issue Reporter)
```

</details>
<details>
<summary>Workspace Info</summary>

```
|  Window (version_2.py - recovery - Visual Studio Code)
|    Folder (recovery): more than 73574 files
|      File types: gif(3679) xml(1330) png(1022) pyi(866) dll(643) mui(527)
|                  p7x(515) t(256) file(251) svg(147)
|      Conf files:;
```

</details>
<details><summary>Extensions (17)</summary>

Extension|Author (truncated)|Version
---|---|---
copilot|Git|1.128.0
copilot-chat|Git|0.8.0
remotehub|Git|0.62.0
gc-excelviewer|Gra|4.2.58
vscode-dotnet-runtime|ms-|1.8.1
python|ms-|2023.18.0
vscode-pylance|ms-|2023.10.40
jupyter|ms-|2023.9.1102792234
jupyter-keymap|ms-|1.1.2
jupyter-renderers|ms-|1.0.17
vscode-jupyter-cell-tags|ms-|0.1.8
vscode-jupyter-slideshow|ms-|0.1.5
remote-wsl|ms-|0.81.8
azure-repos|ms-|0.38.0
remote-repositories|ms-|0.40.0
azurerm-vscode-tools|msa|0.15.11
perl|ric|2.6.1

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
vswsl492:30256859
vslsvsres303:30308271
vserr242:30382549
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
vscorecescf:30445987
vscod805cf:30301675
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593:30376534
pythonvs932:30410667
py29gd2263:30856252
vsclangdc:30486549
c4g48928:30535728
dsvsc012:30540252
pynewext54:30695312
azure-dev_surveyone:30548225
3biah626:30602489
f6dab269:30613381
vscrp:30673768
2i9eh265:30646982
showlangstatbar:30737416
03d35959:30757346
pythonfmttext:30731395
fixshowwlkth:30771522
showindicator:30805244
pythongtdpath:30769146
i26e3531:30792625
gsofa:30866041
pythonnosmt12:30797651
pythonidxpt:30866567
pythonnoceb:30805159
copilotsettingc:30859502
dsvsc013:30795093
dsvsc014:30804076
diffeditorv2:30821572
dsvsc015:30845448
pyreplss1:30865275
pythonmypyd1:30859726
pythontbext0:30864172

```

</details>

**Comment by gjsjohnmurray** [comment]
Did you get a confirmation dialog like this? If so, how did you respond to it?

![image](https://github.com/microsoft/vscode/assets/6726799/00363548-e865-4ea7-a1fb-5383cc501dc1)

**Comment by dbrae** [comment]
I tried cancelling and undoing the connection but was unable to disconnect the repository after clicking on the top button of the 2 in the source control tab. I restarted but was still unable to disconnect. I was under the assumption it would permanently remove them from being attached to my vs code/repository but it turns out it's for the drive on my pc. I had to use easeUS to recover what I could, system restore and recuva didn't help. most my larger files ended up being corrupted. Would running check disk or something similar help with fixing the corrupted files or should I just reformat and move on with what I have recovered already?
